engine/src/main/java/io/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -199,50 +202,27 @@ public void onChildActivating(
             failure -> incidentBehavior.createIncident(failure, childContext));
   }
 
-  private void setLoopVariables(
-      final ExecutableMultiInstanceBody multiInstanceBody,
-      final BpmnElementContext childContext,
-      final int loopCounter,
-      final DirectBuffer inputElement) {
-
-    final var loopCharacteristics = multiInstanceBody.getLoopCharacteristics();
-    loopCharacteristics
-        .getInputElement()
-        .ifPresent(
-            variableName ->
-                stateBehavior.setLocalVariable(childContext, variableName, inputElement));
-
-    // Output multiInstanceBody expressions that are just a variable or nested property of a
-    // variable need to
-    // be initialised with a nil-value. This makes sure that they are not written at a non-local
-    // scope.
-    loopCharacteristics
-        .getOutputElement()
-        .flatMap(Expression::getVariableName)
-        .map(BufferUtil::wrapString)
-        .ifPresent(
-            variableName -> stateBehavior.setLocalVariable(childContext, variableName, NIL_VALUE));
-
-    stateBehavior.setLocalVariable(
-        childContext, LOOP_COUNTER_VARIABLE, wrapLoopCounter(loopCounter));
-  }
-
```

Comment:
Normally this is hard to review, but we already talked about it (automated method reordering). Up to you whether you still want to move it into another commit

Reply:
won't do